src/components/input/input-editable.jsx

```diff
@@ -0,0 +1,158 @@
+  }
+
+  onEdit = () => {
+    this.setState({ originalValue: this.state.value, editing: !this.state.editing });
+  };
+
+  onFocus = event => {
+    if (this.props.onFocus) {
+      this.props.onFocus(event);
+    }
+  };
+
+  onBlur = event => {
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  };
+
+  onChange = event => {
+    this.setState({ value: event.target.value || this.props.emptyDefaultValue });
+
+    if (this.props.onChange) {
+      this.props.onChange(event);
+    }
+  };
+
+  onSubmit = async event => {
+    event.preventDefault();
+    if (!this.props.hasError) {
+      if (this.props.onSubmit) {
+        this.setState({ saving: true });
+        await this.props.onSubmit(this.state.value);
+        this.setState({ saving: false });
+      }
+      this.setState({ editing: !this.state.editing });
+    }
+  };
+
+  onCancel = () => {
+    this.setState({ value: this.state.originalValue });
+    if (this.props.onCancel) {
+      this.props.onCancel(this.state.originalValue);
+    }
+    this.setState({ editing: !this.state.editing });
+  };
+
+  render() {
+    const { classes } = this.props;
+
+    const renderInput = this.state.editing ? (
+      <form className={classes.form} onSubmit={this.onSubmit}>
+        <Input
+          autoFocus
+          className={classes.input}
+          style={this.props.style}
+          variant={this.props.variant === 'primary' ? 'secondary' : 'primary'}
+          disabled={this.props.disabled}
+          label={this.props.label}
+          placeholder={this.props.placeholder}
+          id={this.props.id}
+          value={this.state.value}
+          onFocus={this.onFocus}
+          onBlur={this.onBlur}
+          onChange={this.onChange}
+          valueFormatter={this.props.valueFormatter}
+          hasSuccess={this.props.hasSuccess}
+          hasError={this.props.hasError}
+          hasWarning={this.props.hasWarning}
+          helpText={this.props.helpText}
+          leftAddon={this.props.leftAddon}
+          rightAddon={this.props.rightAddon}
+        />
+      </form>
+    ) : (
+      <span className={classes.readOnly} style={this.props.style}>
+        {this.state.value}
+      </span>
+    );
+
+    const renderButtons = this.state.editing ? (
+      <div className={classes.buttons}>
+        <Button variant="primary" modifier="action" onClick={this.onSubmit} disabled={this.props.hasError}>
+          {this.state.saving ? <Spinner size={13} color="#FFF" /> : this.props.submitLabel}
+        </Button>
+        <Button variant="secondary" modifier="action" onClick={this.onCancel}>
+          {this.props.cancelLabel}
+        </Button>
+      </div>
+    ) : (
+      <Button className={classes.buttonEdit} onClick={this.onEdit} variant="secondary" icon={<Icon.Edit />} />
+    );
+
+    return (
+      <div className={classNames(classes.inputEditable, this.props.className)}>
+        {renderInput}
+        {renderButtons}
+      </div>
```

Comment:
nitpick: just call them "input" and "buttons" or something instead? prefixing "render" implies a function to be ran in my mind

Reply:
Done! :)